File: model/modules/LAI/LAI_forcing/LAI_forcing.m
function [f,fe,fx,s,d,p]=LAI_forcing(f,fe,fx,s,d,p,info,tix)
% +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
% sets the value of s.cd.LAI from the forcing in every time step
%
% Inputs:
%   - f.LAI read from the forcing data set
%
% Outputs:
%   - s.cd.LAI: the value of LAI for current time step
%
% Modifies:
%   - s.cd.LAI
%
% References:
%   - 
%
% Created by:
%   - Sujan Koirala (skoirala)
%
% Versions:
%   - 1.0 on 11.11.2019 (skoirala): moved LAI from d.LAI.LAI to s.cd.LAI
%
% +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

%% 
s.cd.LAI = f.LAI(:,tix);
end

File: src/Models/LAI/LAI_forcing.jl
export LAI_forcing

struct LAI_forcing <: LAI end

function compute(params::LAI_forcing, forcing, land, helpers)
    ## unpack forcing
    @unpack_nt f_LAI ⇐ forcing

    LAI = f_LAI
    ## pack land variables
    @pack_nt LAI ⇒ land.states
    return land
end

purpose(::Type{LAI_forcing}) = "sets the value of land.states.LAI from the forcing in every time step"

@doc """

$(getBaseDocString(LAI_forcing))

---

# Extended help

*References*

*Versions*
 - 1.0 on 11.11.2019 [skoirala]: moved LAI from land.LAI.LAI to land.states.LAI  

*Created by:*
 - skoirala
"""
LAI_forcing
